Assets/Scripts/Environment/FrustumCulling.cs

```diff
@@ -36,7 +36,23 @@ private void UpdateCullAbles()
             
             for (int i = 0; i < length; i++)
             {
-                _cullAbleObjects[i].SetActive(GeometryUtility.TestPlanesAABB(planes, _cachedRenderers[i].bounds));
+                bool isVisible = GeometryUtility.TestPlanesAABB(planes, _cachedRenderers[i].bounds);
+                if (isVisible)
+                {
+                    // Cast a ray from the camera to the object's center
+                    Vector3 direction = _cullAbleObjects[i].transform.position - _this.transform.position;
+                    float distance = direction.magnitude;
+                    RaycastHit hit;
+                    if (Physics.Raycast(_this.transform.position, direction, out hit, distance))
+                    {
+                        // If the ray hits something other than the object itself, it's not visible
+                        if (hit.transform!= _cullAbleObjects[i].transform)
+                        {
+                            isVisible = false;
+                        }
+                    }
+                }
+                _cullAbleObjects[i].SetActive(isVisible);
```

Comment:
```suggestion
                }
                
                _cullAbleObjects[i].SetActive(isVisible);
```
[code convention](https://github.com/Team-Swamp/Codename-BDLMTW/wiki/Code-convention#if-statement-and-loop)